ext/ldap/ldap.stub.php

```diff
@@ -2,265 +2,151 @@
 
 #ifdef HAVE_LDAP_SASL
-/** @param resource $ldap */
-function ldap_sasl_bind($ldap, ?string $dn = null, ?string $password = null, ?string $mech = null, ?string $realm = null, ?string $authc_id = null, ?string $authz_id = null, ?string $props = null): bool {}
+function ldap_sasl_bind(LDAP $ldap, ?string $dn = null, ?string $password = null, ?string $mech = null, ?string $realm = null, ?string $authc_id = null, ?string $authz_id = null, ?string $props = null): bool {}
 #endif
 
-/**
- * @param resource|array $ldap
- * @return resource|array|false
- */
-function ldap_read($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, ?array $controls = null) {}
+/** @param LDAP|array $ldap */
+function ldap_read($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, ?array $controls = null): LDAPResult|array|false {}
 
-/**
- * @param resource|array $ldap
- * @return resource|array|false
- */
-function ldap_list($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, ?array $controls = null) {}
+/** @param LDAP|array $ldap */
+function ldap_list($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, ?array $controls = null): LDAPResult|array|false {}
 
-/**
- * @param resource|array $ldap
- * @return resource|array|false
- */
-function ldap_search($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, ?array $controls = null) {}
-
-/** @param resource $ldap */
-function ldap_free_result($ldap): bool {}
+/** @param LDAP|array $ldap */
+function ldap_search($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, ?array $controls = null): LDAPResult|array|false {}
 
+function ldap_free_result(LDAPResult $result): bool {}
 
-/**
- * @param resource $ldap
- * @param resource $result
- */
-function ldap_count_entries($ldap, $result): int {}
+function ldap_count_entries(LDAP $ldap, LDAPResult $result): int {}
 
-/**
- * @param resource $ldap
- * @param resource $result
- * @return resource|false
- */
-function ldap_first_entry($ldap, $result) {}
+function ldap_first_entry(LDAP $ldap, LDAPResult $result): LDAPResultEntry|false {}
 
-/**
- * @param resource $ldap
- * @param resource $result
- * @return resource|false
- */
-function ldap_next_entry($ldap, $result) {}
+function ldap_next_entry(LDAP $ldap, LDAPResultEntry $entry): LDAPResultEntry|false {}
```

Comment:
the 2nd parameter name used to be buggy. I can fix it in 8.0 as well.

Reply:
Yeah, please fix it in 8.0.